### ISSUE TITLE
Custom bundles should not contain unwanted debug-resources

### ISSUE THREAD

**Issue by RandomByte** [SAP/ui5-tooling#513]
## Expected Behavior
Custom bundles should never contain debug files generated during the build of dependencies.

To be checked: What about "debug bundles"?

## Current Behavior
Custom bundles using wildcards in their configuration might include debug resources of dependencies if dependencies are built as well (`--all`).

## Steps to Reproduce the Issue

1. Clone the [openui5-sample-app](https://github.com/SAP/openui5-sample-app)
2. Add the following configuration to the `ui5.yaml`
    ```yaml
    builder:
      bundles:
        - bundleDefinition:
            name: my-bundle.js
            sections:
              - mode: provided
                filters: []
              - mode: preload
                name: bundle
                filters:
                  - sap/ui/core/
    ```

3. Execute `ui5 build --all --clean-dest`
4. ⚠️ The generated bundle at `dist/resources/my-bundle.js` contains debug files of the sap/ui/core namespace like `sap/ui/core/AppCacheBuster-dbg.js`

## Context
 * UI5 Module Version (output of `ui5 --version` when using the CLI): **`2.10.2`**
 * Node.js Version: **`v15.12.0`**
 * npm Version: `7.6.3`
 * OS/Platform: `macOS`
 * Browser *(if relevant)*: `-`
 * Other information regarding your environment *(optional)*: `nothing`

## Log Output / Stack Trace

```
{...}
```

**Comment by RandomByte** [comment]
Additional info from today's sync: The Maven bundling filters out debug resources for which a non-debug version can be found, unless explicitly requested to prefer debug resources.

**Comment by RandomByte** [comment]
Tracking internally as CPOUI5FOUNDATION-333

**Comment by RandomByte** [comment]
Fixed in `next` (UI5 Tooling 3.0) via https://github.com/SAP/ui5-builder/pull/666
